ocs_ci/ocs/fiojob.py

```diff
@@ -548,6 +551,10 @@ def is_storage_reclaimed():
         # Without checksum, we just need to make sure that data were deleted
         # and wait for this to happen to avoid conflicts with tests executed
         # right after this one.
-        delete_fio_data(fio_job_file, is_storage_reclaimed)
+        if delete_fio_data_after_test:
+            delete_fio_data(fio_job_file, is_storage_reclaimed)
+        # Wait until the end of the tests, and then delete the data.
+        else:
+            request.addfinalizer(lambda: delete_fio_data(fio_job_file, is_storage_reclaimed))
```

Comment:
Let me think whether this is a good idea or not.

Reply:
Ok, so we have 2 options here:

- drop both `delete_fio_data_after_test` and `request` options, and use something like `keep_fio_data` option instead, so that when this is True, `delete_fio_data()` is not executed
- move execution of  `delete_fio_data()` to teardown phase as done now, but simplify the interface (instead of  `delete_fio_data_after_test` and `request`, use single option `teardown_request` with explanation that when you pass a pytest request object there, the data will be explicitly deleted during test case teardown. But for this to work in all cases, we need to make sure that `max_avail_before_delete` is also populated during teardown.

Otherwise I like this.

Reply:
I discussed this with Itzhak and we agreed to with 1st option, because it seems to work good enough for this new usecase and requires minimal change for this PR to work.